content/intro-to-reading-data.md

```diff
@@ -48,10 +48,11 @@ Every interaction with the Solana network using `@solana/web3.js` is going to ha
 
 For now we'll use the `Devnet` cluster rather than `Mainnet`. As the name suggests, the `Devnet` cluster is designed for developer use and testing.
 
+We further specify that each such interaction must be `Confirmed` by the whole cluster before we move forward at each step.
 ```typescript
 import { Connection, clusterApiUrl } from "@solana/web3.js";
 
-const connection = new Connection(clusterApiUrl("devnet"));
+const connection = new Connection(clusterApiUrl("devnet"), "confirmed");
```

Comment:
@mikemaccana I haven't checked in a minute, but is this necessary for these code snippets to work?

Reply:
Will do a quick check...

Reply:
Sorry for the wait over the holidays @BrainyZombie. This seems a little odd - per [the web3.js docs](https://docs.solana.com/api/http#default)

> If commitment configuration is not provided, the node will default to "finalized" commitment

Which is **more** final than `confirmed` is. 

You mentioned module 2, lesson 1 was failing - there's a lot of code in that lesson (we're trying to do smaller chunks these days) - could you please provide an isolated reproduction case so we can investigate and fix this?

Reply:
@mikemaccana I believe this code reproduces the issue. https://codesandbox.io/p/devbox/solana-reproduction-4h6skq
I am running it in a loop since it doesn't break every time when running once. The loop can be removed and it will break once every few times it's run.

This may be a bug with solana's JS api. Adding 'confirmed' makes it work though.

https://github.com/solana-labs/solana-program-library/issues/3326
https://stackoverflow.com/questions/76445810/tokenaccountnotfounderror-encountered-while-trying-to-create-token-account

I did not realize that web3.js is supposed to be waiting for finalize by default, so this bug may not exist in other functions, and may only need to be worked around in this particular case.

Reply:
it's useful, i resolved by this method, but if it's production envoriment, add "confirmed" state is ok?